### PR TITLE
Print retrieved documents counts

### PR DESCRIPTION
**Pull request by we-ai**
This PR adds logs showing counts of retrieved documents from Firestore, as described in [issue 1016](https://github.com/episphere/connect/issues/1016). Some code clean-ups are also done.

### CODE REVIEW

File: utils/firestore.js [hunk 1]
```diff
@@ -111,20 +113,6 @@ const updateResponse = async (data, uid) => {
     }
 }
 
-const incrementCounter = async (field, siteCode) => {
```
Comment (anthonypetersen) [inline utils/firestore.js:114]: assuming these weren't being used anywhere in the code base?

File: utils/firestore.js [hunk 2]
```diff
@@ -323,41 +312,41 @@ const retrieveParticipants = async (siteCode, decider, isParent, limit, page, si
...
 }
 
+// TODO: Avoid using `offset` for pagination, because offset documents are still read and charged.
 const retrieveRefusalWithdrawalParticipants = async (siteCode, isParent, concept, limit, page) => {
     try {
         const operator = isParent ? 'in' : '==';
         const offset = (page - 1) * limit;
         
-        let participants = await db.collection('participants')
+        const snpashot = await db.collection('participants')
```
Comment (anthonypetersen) [inline utils/firestore.js:333]: spell check

File: utils/firestore.js [hunk 3]
```diff
@@ -367,7 +356,7 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
         const { incentiveConcepts } = require('./shared');
         const object = incentiveConcepts[roundType]
         
-        let participants = await db.collection('participants')
+        const snpashot = await db.collection('participants')
```
Comment (anthonypetersen) [inline utils/firestore.js:359]: spell check

File: utils/firestore.js [hunk 4]
```diff
@@ -323,41 +312,41 @@ const retrieveParticipants = async (siteCode, decider, isParent, limit, page, si
...
                                 .where('827220437', operator, siteCode)
                                 .where(concept, '==', 353358909)
                                 .orderBy('Connect_ID', 'asc')
                                 .offset(offset)
                                 .limit(limit)
                                 .get();                 
+        printDocsCount(snpashot, "retrieveRefusalWithdrawalParticipants");
```
Comment (anthonypetersen) [inline utils/firestore.js:340]: spell check

File: utils/firestore.js [hunk 5]
```diff
@@ -323,41 +312,41 @@ const retrieveParticipants = async (siteCode, decider, isParent, limit, page, si
...
 
-        return participants.docs.map(document => {
-            return document.data();
-        });
+        return snpashot.docs.map(doc => doc.data());
```
Comment (anthonypetersen) [inline utils/firestore.js:342]: spell check

File: utils/firestore.js [hunk 6]
```diff
@@ -377,8 +366,9 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
                                 .offset(offset)
                                 .limit(limit)
                                 .get();
-                        
-        return participants.docs.map(document => {
+        printDocsCount(snpashot, "retrieveParticipantsEligibleForIncentives");
```
Comment (anthonypetersen) [inline utils/firestore.js:369]: spell check

File: utils/firestore.js [hunk 7]
```diff
@@ -377,8 +366,9 @@ const retrieveParticipantsEligibleForIncentives = async (siteCode, roundType, is
...
+
+        return snpashot.docs.map(document => {
```
Comment (anthonypetersen) [inline utils/firestore.js:371]: spell check

File: utils/firestore.js [hunk 8]
```diff
@@ -509,10 +501,11 @@ const removeUninvitedParticipants = async () => {
 
         while (willContinue) {
             const currSnapshot = await db
```
Comment (anthonypetersen) [inline utils/firestore.js:503]: change to snapshot?

File: utils/firestore.js [hunk 9]
```diff
@@ -1348,6 +1372,7 @@ const getSpecimenAndParticipant = async (collectionId, siteCode, isBPTL) => {
         let query = db.collection('biospecimen').where(fieldMapping.collectionId.toString(), '==', collectionId);
         if (!isBPTL) query = query.where(fieldMapping.healthCareProvider.toString(), '==', siteCode);
         const specimenSnapshot = await query.get();
+        printDocsCount(specimenSnapshot, "getSpecimenAndParticipant; collection: biospecimen");
```
Comment (anthonypetersen) [inline utils/firestore.js:1375]: snapshot?

File: utils/firestore.js [hunk 10]
```diff
@@ -1799,6 +1841,7 @@ const getNotificationSpecifications = async (notificationType, notificationCateg
     }
 }
 
+// TODO: Avoid using `offset` for pagination, because offset documents are still read and charged.
```
Comment (anthonypetersen) [inline utils/firestore.js:1844]: can you explain this? Example, say I have 1000 documents and my limit is 100. If I'm using `offset` for pagination, are you saying we will in the end be charged for reading more than 1000 documents?

File: utils/firestore.js [hunk 11]
```diff
@@ -2352,6 +2410,7 @@ const assignKitToParticipant = async (data) => {
 const processVerifyScannedCode = async (id) => {
     try {
         const snapShot = await db.collection('kitAssembly').where('531858099', '==', id).where('221592017', '==', 241974920).get();
```
Comment (anthonypetersen) [inline utils/firestore.js:2412]: snapshot*

File: utils/firestore.js [hunk 12]
```diff
@@ -2752,6 +2821,7 @@ const setPackageReceiptFedex = async (boxUpdateData) => {
         }
         
         const boxSnapshot = await query.get();
```
Comment (anthonypetersen) [inline utils/firestore.js:2823]: snapshot*

**Review by anthonypetersen — APPROVED** [review]
everything looks structurally fine... found a few spell checks, found a few instances you weren't using `snapshot` when it appeared that was the consistency you were looking for, also had a question on your comment

File: utils/firestore.js (hunk 1 above)
Comment (we-ai) [reply utils/firestore.js:114]: It's not, so I removed it.

File: utils/firestore.js (hunk 10 above)
Comment (we-ai) [reply utils/firestore.js:1844]: I believe so: the offset docs are also counted and charged. I'm now writing based on what I understand, so the below data may not be accurate.
In the example mentioned, if pagination using `offset`  to get all the 1000 documents and 100 docs retrieved per pagination, the total count of reads will be 100 + 200 + 300 +.. + 1000 = 5500.

Comment (anthonypetersen) [reply utils/firestore.js:1844]: This is something definitely worth looking into / verifying... We use offset for pagination in our APIs that return the largest amount of data, such as `getParticipants`, if what you're saying is correct then it's adding way more than we expect to our read document count.

Comment (we-ai) [reply utils/firestore.js:1844]: Here is documentation explaining why `offset` should be avoided when reading large datasets: https://firebase.google.com/docs/firestore/pricing

**Review by jhflorey — APPROVED** [review]

**Pull request merged**